### PR TITLE
Setup SAAT and SIDAM

### PR DESCRIPTION
**Pull request by tchow8**
[DIV-4139](https://tools.hmcts.net/jira/browse/DIV-4319)

Added Serenity Reporting and expose to Jenkins.
Change SAAT IDAM url to use Strategic
Dynamically change expected document url based on current document management store url in use.

### CODE REVIEW

**Comment by tchow8** [comment]
Closing in fvour of https://github.com/hmcts/div-case-data-formatter/pull/110

File: src/integrationTest/java/uk/gov/hmcts/reform/divorce/model/UserGroup.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package uk.gov.hmcts.reform.divorce.model;
+
+import lombok.Builder;
+import lombok.Getter;
+
+@Builder
+@Getter
+public class UserGroup {
```
Comment (qzhou-hmcts) [inline src/integrationTest/java/uk/gov/hmcts/reform/divorce/model/UserGroup.java]: Builder for one attribute?
You can use @Value to create  immutable  class inject the value in the constructor

File: src/integrationTest/java/uk/gov/hmcts/reform/divorce/ResourceLoader.java [hunk 2]
```diff
@@ -20,4 +22,12 @@ static String loadJson(final String filePath) throws Exception {
 
         return Files.readAllBytes(Paths.get(url.toURI()));
     }
+
+    public static <T> String objectToJson(T object) {
+        try {
+            return new ObjectMapper().writeValueAsString(object);
```
Comment (qzhou-hmcts) [inline src/integrationTest/java/uk/gov/hmcts/reform/divorce/ResourceLoader.java]: Could you crate a static instance of ObjectMapper so It won't be  instantiate each time it's used

File: src/integrationTest/java/uk/gov/hmcts/reform/divorce/model/UserGroup.java (hunk 1 above)
Comment (tchow8) [reply src/integrationTest/java/uk/gov/hmcts/reform/divorce/model/UserGroup.java]: done

File: src/integrationTest/java/uk/gov/hmcts/reform/divorce/ResourceLoader.java (hunk 2 above)
Comment (tchow8) [reply src/integrationTest/java/uk/gov/hmcts/reform/divorce/ResourceLoader.java]: done

**Pull request merged**